pkg/plugins/handlers.go

```diff
@@ -26,6 +26,79 @@ func NewPluginsHandler(client *http.Client, token string, pluginsEndpointMap map
 	}
 }
 
+func (p *PluginsHandler) HandleLocales(w http.ResponseWriter, r *http.Request) {
+	if r.Method != "GET" {
+		w.Header().Set("Allow", "GET")
+		serverutils.SendResponse(w, http.StatusMethodNotAllowed, serverutils.ApiError{Err: "Method unsupported, the only supported methods is GET"})
+		return
+	}
+
+	lang, pluginLocaleFile, err := parseLangAndPluginLocaleFile(r.URL.Path)
+	if err != nil {
+		errMsg := err.Error()
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	redirectTo := fmt.Sprintf("static/locales/%s/%s", lang, pluginLocaleFile)
+	pluginName := strings.SplitN(pluginLocaleFile, ".", 2)[0]
+	for _, enabledPluginName := range p.PluginsEndpointMap {
+		// if the request if for enabled plugin locale, then proxy to it's service
+		if enabledPluginName == pluginName {
+			pluginServiceEndpoint, err := p.getServiceRequestURL(pluginName)
+			if err != nil {
+				errMsg := err.Error()
+				klog.Error(errMsg)
+				serverutils.SendResponse(w, http.StatusBadRequest, serverutils.ApiError{Err: errMsg})
+				return
+			}
+			redirectTo = fmt.Sprintf("%s/%s/%s", pluginServiceEndpoint, lang, pluginLocaleFile)
+		}
+		klog.Errorf(redirectTo)
+	}
+
+	resp, err := p.Client.Get(redirectTo)
+	if err != nil {
+		errMsg := fmt.Sprintf("GET request for %q locale file failed: %v", pluginLocaleFile, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		errMsg := fmt.Sprintf("GET request for %q locale file failed with %d status code", pluginLocaleFile, resp.StatusCode)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, resp.StatusCode, serverutils.ApiError{Err: errMsg})
+		return
+	}
+
+	// filter unwanted headers from the response
+	proxy.FilterHeaders(resp)
+	for key, value := range resp.Header {
+		for _, v := range value {
+			w.Header().Add(key, v)
+		}
+	}
+
+	_, err = io.Copy(w, resp.Body)
+	if err != nil {
+		errMsg := fmt.Sprintf("failed sending HTTP response body from %q locale file: %v", pluginLocaleFile, err)
+		klog.Error(errMsg)
+		serverutils.SendResponse(w, http.StatusBadGateway, serverutils.ApiError{Err: errMsg})
+		return
+	}
+}
+
+// parseLangAndPluginName will parse the language and plugin locale file
+func parseLangAndPluginLocaleFile(urlPath string) (string, string, error) {
+	langAndPluginLocaleFile := strings.SplitN(urlPath, "/", 2)
```

Comment:
I wish we were using something like https://github.com/gorilla/mux to handle variables in the path, although that seems like overkill for one endpoint.

I'd suggest simply using query parameters `lng` and `ns` to avoid having to parse the path. We might eventually support multi-loading, and the query parameters would fit better with that (although that could be difficult if we need to proxy some requests).

https://github.com/i18next/i18next-http-backend#backend-options